tests/Utils/SchemaExtenderTest.php

```diff
@@ -352,26 +391,32 @@ enum SomeEnum {
      */
     public function testExtendsUnionsByAddingNewTypes(): void
     {
-        $extendedSchema = $this->extendTestSchema('
-          extend union SomeUnion = Bar
-        ');
-        self::assertSame(
-            $this->printTestSchemaChanges($extendedSchema),
-            <<<'EOF'
-union SomeUnion = Foo | Biz | Bar
+        $schema = BuildSchema::build('
+      type Query {
+        someUnion: SomeUnion
+      }
 
-EOF
-        );
+      union SomeUnion = Foo | Biz
 
-        $queryType = $extendedSchema->getQueryType();
-        self::assertInstanceOf(ObjectType::class, $queryType);
+      type Foo { foo: String }
+      type Biz { biz: String }
+      type Bar { bar: String }
+        ');
+        $extendedSchema = SchemaExtender::extend($schema, Parser::parse('
+          extend union SomeUnion = Bar
+        '));
 
+        self::assertSame([], $schema->validate());
         self::assertSame(
-            $queryType->getField('someUnion')->getType(),
-            $extendedSchema->getType('SomeUnion')
+            <<<'EOF'
+union SomeUnion = Foo | Biz | Bar
+EOF,
+            self::printSchemaChanges($schema, $extendedSchema),
         );
     }
 
+    // TODO continue converting tests from here
```

Comment:
TODO finish before merging.

Reply:
After #1139, we will be much closer to merging #928 and #929; it should help you here (all these tests have already been converted there).

Reply:
Thank you, I included https://github.com/webonyx/graphql-php/pull/929

Reply:
@spawnia That's definitely not what I meant! It makes this PR huge and impossible to review. Which changes are to solve the eager-loading problem and which are to address the ordering issue? Impossible to say.

`tests/Utils/SchemaExtenderLegacyTest.php` shouldn't be included. It has been merged to master in #977, addressed in consequent PRs, and finally removed in #1111. We can only guess how many other merge conflicts were resolved incorrectly… I have been working for months to get #929 (and #928) merged nicely and clearly and you do what with it? A big ball of mud…